File: mario_pyramid/Python/mario.py
print("Mario Pyramid in Python!")

# Prints mario pyramid given a size
def mario_pyramid(size):

	row = "#"

	# Prints row of blocks
	for block in range(size):
		print(row)
		row+="#"

mario_pyramid(10)

# Prints mario pyramid in reverse
def mario_reverse_pyramid(size):

	row = "#"

	# Prints row of Blocks
	for block in range(size):
		# Prints spaces
		spaces = ""	
		for space in range(size - len(row)):
			spaces += " "
			
		print(spaces+row);
		row += "#"

mario_reverse_pyramid(10)

File: mario_pyramid/Ruby/mario.rb
puts "Mario Pyramid in Ruby!"

# Prints Mario Pyramid
def mario_pyramid size

	row = "#"

	# Prints rows of blocks
	for block in 0...size
		puts row
		row += "#"
	end

end

mario_pyramid 10


# Prints reverse Mario pyramid
def mario_reverse_pyramid size

	row = "#"

	# Blocks
	for block in 0...size

		spaces = ""

		# Spaces
		for space in 0...(size-row.length)

			spaces += " "
		end

		puts spaces + row

		row += "#"
	end

end


mario_reverse_pyramid 10